src/components/ActionNetworkWarning.jsx

```diff
@@ -1,30 +1,59 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import NetworkChangeGuide from './NetworkChangeGuide';
+import { Button } from 'antd';
+import Web3 from 'web3';
+import config from '../configuration';
 
 /**
  * It is used to show in modals when user wants to take an action while connected to wrong network.
  */
 
 function ActionNetworkWarning(props) {
-  const { incorrectNetwork, networkName } = props;
+  const { incorrectNetwork, networkName, switchNetwork, web3, isInline } = props;
+  const isMetaMask = web3 && web3.MetaMask;
+
+  const switchNetworkBtn = () =>
+    // Just MetaMask supports changing network
+    isMetaMask && (
+      <div className="my-2">
+        <Button type="primary" ghost className="ml-4 rounded" onClick={switchNetwork}>
+          Switch Network
+        </Button>
+      </div>
+    );
+
   return (
     incorrectNetwork && (
-      <div>
-        <div className="alert alert-warning">
-          <i className="fa fa-exclamation-triangle" />
-          To take this action you need to connect metamask to <strong>{networkName}</strong>{' '}
-          network. Then try again.
+      <div className={isInline ? '' : 'text-center'}>
+        <div className="alert alert-warning d-flex flex-wrap align-items-center">
+          <div className="d-flex align-items-center">
+            <i className="fa fa-exclamation-triangle mr-3" />
+            <div>
+              To enable all actions, please connect your wallet to <strong>{networkName}</strong>{' '}
+              network.
+            </div>
+          </div>
+          {isInline && switchNetworkBtn()}
         </div>
-        {NetworkChangeGuide}
+        {!isInline && switchNetworkBtn()}
       </div>
     )
   );
 }
 
 ActionNetworkWarning.propTypes = {
   incorrectNetwork: PropTypes.bool.isRequired,
-  networkName: PropTypes.string.isRequired,
+  networkName: PropTypes.string,
+  switchNetwork: PropTypes.func,
```

Comment:
Since `switchNetwork` is constant (the one obtained from web3 provider), we can remote this parameter

Reply:
I can't use web3provider context here because it causes a dependency cycle

![image](https://user-images.githubusercontent.com/73441857/135134370-e007cc7e-bc0d-4fd2-81e2-cbf829e4cfea.png)